File: src/main/java/com/sothawo/springdataelastictest/presidents/PresidentRepository.java
/*
 * (c) Copyright 2020 codecentric AG
 */
package com.sothawo.springdataelastictest.presidents;

import org.springframework.data.elasticsearch.core.SearchHit;
import org.springframework.data.elasticsearch.repository.ReactiveElasticsearchRepository;
import reactor.core.publisher.Flux;

public interface PresidentRepository extends ReactiveElasticsearchRepository<President, String> {
    Flux<SearchHit<President>> searchByTerm(Integer year);
}

File: src/main/kotlin/com/sothawo/springdataelastictest/presidents/PresidentRepository.kt
package com.sothawo.springdataelastictest.presidents

import kotlinx.coroutines.flow.Flow
import org.springframework.data.elasticsearch.core.SearchHit
import org.springframework.data.elasticsearch.repository.CoroutineElasticsearchRepository

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
interface PresidentRepository : CoroutineElasticsearchRepository<President, String> {
    suspend fun searchByTerm(year: Int): Flow<SearchHit<President>>
}
